jina/flow/__init__.py

```diff
@@ -790,6 +790,56 @@ def my_reader():
         """
         self._get_client(**kwargs).search(input_fn, output_fn, **kwargs)
 
+    def plot(self, **kwargs) -> 'Flow':
+        """
+            Output the mermaid graph for visualization
+            :return: a mermaid-formatted string
+            """
+        mermaid_graph = list()
+        for node, v in self._pod_nodes.items():
+            for need in sorted(v.needs):
+                curr_line = need + '[' + need + ']' + ' --> ' + node + '[' + node + ']'
+                mermaid_graph.append(curr_line)
+
+        mermaid_str = 'graph TD\n' + '\n'.join(mermaid_graph)
+        if 'output' in kwargs:
+            self._mermaidstr_to_jpg(mermaid_str=mermaid_str, output=kwargs['output'])
+        else:
+            return self._mermaidstr_to_url(mermaid_str)
+
+        return mermaid_str
```

Comment:
this is not correctly implemented. it does not realize the [fluent pattern](https://en.wikipedia.org/wiki/Fluent_interface) as I mentioned. Please refer to `.add()` and `.needs()` implementation and return a `op_flow` object.

Always remember to add a test for it